code/modules/keybindings/bindings_client.dm

```diff
@@ -20,22 +20,22 @@
 	//don't want to kick people if a lag spike results in a huge flood of commands being sent
 	if(cache >= MAX_KEYPRESS_AUTOKICK)
 		if(!keysend_tripped)
-			keysend_tripped = TRUE		
+			keysend_tripped = TRUE
 			next_keysend_trip_reset = world.time + (2 SECONDS)
 		else
 			log_admin("Client [ckey] was just autokicked for flooding keysends; likely abuse but potentially lagspike.")
 			message_admins("Client [ckey] was just autokicked for flooding keysends; likely abuse but potentially lagspike.")
-			QDEL_IN(src, 1)
+			qdel(src)
```

Comment:
in case you're wondering client does return harddel now

Reply:
if it's harddelling couldn't that potentially be a vector as well? Feels exploitable.

Reply:
assuredly it would be hard to join fast enough to do that.

Reply:
With a single account yeah, but we're not dealing with standard unmodified clients with normal limitations now.